### ISSUE TITLE
Update Terraform Provider RKE to 1.3.11

### ISSUE THREAD

**Issue by annablender** [rancher/terraform-provider-rke#342]
Bump RKE version to 1.3.11

This version uses etcd v3.5.3 that includes the fix for https://github.com/etcd-io/etcd/issues/13766. Due to this etcd bug, versions 1.3.9 was not recommended for production use.

I'm far form being a golang guru so I based my changes mainly on @VIKMSTR and @fe-ax already opened PRs. Since 1.3.10 ships with the etcd fix, I think we should concentrate on this PRs if that's OK for you @VIKMSTR and @fe-ax. I successfully built the provider and used it to build a test cluster with 3 control planes nodes and 2 workers. Still, I may have missed some key changes that should be made, so double checking would be wise ^^.

Changes include
- Update Go version to 1.17.9 (1.17 is the version used by RKE)
- Updating dependencies in go.mod. I updated all deps to their latest available version except for github.com/hashicorp/terraform-plugin-sdk. I didn't upgrade it to v2 because it seems like many changes in provider code should be made to make it compatible so this should be part of a dedicated PR IMO. Therefore, I simply updated this dep to the last v1.x available
- Force github.com/spf13/afero to stick to v1.2.2 required by the version of github.com/hashicorp/terraform-plugin-sdk mentioned above

**Comment by tsde** [comment]
@annablender Following your [comment](https://github.com/rancher/terraform-provider-rke/pull/340#issuecomment-1132111307), I ran an acceptance test with `make testacc`. The `TestAccResourceRKECluster` test fails on my branch but also on the master branch with the following error:
```
Failed running cluster err:Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system
--- FAIL: TestAccResourceRKECluster (214.07s)
```

Complete log file: [test.log](https://github.com/rancher/terraform-provider-rke/files/8734186/test.log)

**Comment by annablender** [comment]
I also tested `make testacc` in my local terraform RKE branch and ran into errors. Not sure what's going on there. This just needs to be tested to make sure an RKE cluster with all roles (etcd nodes as well) works and the dockerd CRI can be enabled for kubernetes 1.24 and the cluster still works as expected. Please run those tests after https://github.com/rancher/terraform-provider-rke/pull/337#pullrequestreview-977276398 has been merged. Thank you for your patience!

**Comment by tsde** [comment]
@annablender I locally applied the PR #337 and tested the provisionning of a one node cluster v1.23.6 (with etcd/control plane/worker roles). I managed to have a cluster up and running with either `enable_cri_dockerd` set to `false` or `true`.
Of course, I can't test 1.24 as it is not supported by RKE yet, but at least PR #337 seems to do the job.

As a side note, enabling the option led me to the behavior described in https://github.com/rancher/rke/issues/2938 so I added a comment with my findings. This is not related to this terraform provider in any way, but it may be a blocker in a near future for supporting 1.24.

**Comment by annablender** [comment]
Great, thanks for testing this!

**Comment by annablender** [comment]
## Testing template

#### Root cause

Update Terraform RKE provider.
 
#### What was fixed, or what changes have occurred

Two major changes have been done by community PRs (thank you open source!)

* Bump RKE to version 1.3.11 https://github.com/rancher/terraform-provider-rke/blob/b64dcadbb8502fd64108534a1f98592c9e0d03e3/go.mod#L9
* Add `enable_cri_dockerd` parameter for kubernetes 1.24 support https://github.com/rancher/terraform-provider-rke/blob/b64dcadbb8502fd64108534a1f98592c9e0d03e3/rke/schema_rke_cluster.go#L146
* Add `enable_cri_dockerd` parameter to test cases

#### Areas or cases that should be tested

* Run `make test` in the provider repo and verify structure tests pass
* Test provisioning a one node cluster with kubernetes version v1.23.6 (with etcd/control plane/worker nodes). Test that cluster is healthy when `enable_cri_dockerd` is both false and true
Test provisioning with kubernetes version 1.24 (when possible)

locally applied the PR https://github.com/rancher/terraform-provider-rke/pull/337 and tested the provisionning of a one node cluster v1.23.6 (with etcd/control plane/worker roles). I managed to have a cluster up and running with either enable_cri_dockerd set to false or true.
Of course, I can't test 1.24 as it is not supported by RKE yet, but at least PR https://github.com/rancher/terraform-provider-rke/pull/337 seems to do the job.
 
#### What areas could experience regressions ?

Provisioning of standalone RKE clusters where `enable_cri_dockerd: true.`
 
#### Are the repro steps accurate/minimal ?

Yes.

**Comment by tmsdce** [comment]
Thanks for merging the related PRs
@annablender @Josh-Diamond any ETA for cutting a new release integrating these changes ?

**Comment by Josh-Diamond** [comment]
@tmsdce Undergoing testing now 👍

**Comment by Josh-Diamond** [comment]
Ticket #342 - Test Results [pt. 1]

Pending test:
  - Test provisioning with kubernetes version 1.24 - [`currently unsupported`]

With Docker on a single-node instance:

Verified on rancher `v2.6.6`:
1. Run `make test` in the provider repo and verify structure tests pass
2. Fresh install of rancher `v.2.6.6`
3. With Terraform, provision a single-node, all-roles RKE cluster w/ k8s `v1.23.6-rancher1-1`, and setting RKE config `enable_cri_dockerd` to true
4. With Terraform, provision a single-node, all-roles RKE cluster w/ k8s `v1.23.6-rancher1-1`, and setting RKE config `enable_cri_dockerd` to false
5. With Terraform, provision a 3 node [`1 etcd`, `1 cp`, `1 wkr`] RKE cluster w/ k8s `v1.23.6-rancher1-1`, and setting RKE config `enable_cri_dockerd` to true
6. Verified - All tests successful


Screenshots:

`Step 1`
<img width="620" alt="Screen Shot 2022-07-05 at 9 56 23 AM" src="https://user-images.githubusercontent.com/46494969/177391767-b0b72a06-cdd3-44e7-a696-1487ebc6c4f4.png">

`Step 3`
<img width="1645" alt="Screen Shot 2022-07-05 at 10 28 48 AM" src="https://user-images.githubusercontent.com/46494969/177391220-021cacd0-4fd8-4db3-bade-2fa30c89e920.png">

`Step 4`
<img width="1645" alt="Screen Shot 2022-07-05 at 10 40 15 AM" src="https://user-images.githubusercontent.com/46494969/177391248-e4a19a92-70b0-4ec1-b58b-c176abf0284c.png">

`Step 5`
<img width="1640" alt="Screen Shot 2022-07-05 at 11 09 05 AM" src="https://user-images.githubusercontent.com/46494969/177391280-81f1f685-d7cd-488d-a971-c20cc5fce812.png">


Additional Information:
  - moving to `blocked` until k8s v1.24 is supported by RKE;
  - will resume testing then

**Comment by Josh-Diamond** [comment]
@tsde Test case surrounding `Test provisioning with kubernetes version 1.24 (when supported by RKE)`, i had mistakenly put on hold, as i was unaware that k8s RKE 1.24 was ready to-test.  Resuming testing now;

**Comment by Josh-Diamond** [comment]
Ticket #342 - Test Results [pt. 2]

With Docker on a single-node instance:

Verified on rancher `v2.6-a678d0e12fe21936f09ae0e906bc420e43d7d35c-head`:
1. Fresh install of rancher `v2.6-head`
2. With Terraform, provision a 3 node [1 etcd, 1 cp, 1 wkr] RKE cluster w/ k8s `v1.24.2-rancher1-1`, and setting RKE config `enable_cri_dockerd` to true
3. Verified - Cluster successfully provisions

Screenshot:
<img width="1642" alt="Screen Shot 2022-07-07 at 9 53 39 AM" src="https://user-images.githubusercontent.com/46494969/177829123-a8e08239-ea52-433d-8b07-afe59e1b3e2d.png">

**Comment by Josh-Diamond** [comment]
Testing was conducted with TF rancher2 provider and NOT TF rke provider;
This was by mistake.  Re-opened to re-test properly

**Comment by Josh-Diamond** [comment]
Ticket #342 - Test Results [pt. 3]

Veri-FAILED with TF RKE `v1.3.1`:
1. Provision a single-node RKE cluster with kubernetes version `v1.23.6`
2. Veri-FAILED - Incompatible versioning; error states `expected kubernetes_version to be one of [...], got v1.23.6-rancher1-1` 

Related reported issue:
- https://github.com/rancher/terraform-provider-rke/issues/349

Full error message:
```logs
Error: expected kubernetes_version to be one of [v1.22.4-rancher1-1 v1.21.7-rancher1-1 v1.21.6-rancher1-2 v1.21.6-rancher1-1 v1.21.5-rancher1-1 v1.21.4-rancher1-1 v1.20.13-rancher1-1 v1.20.12-rancher1-2 v1.20.12-rancher1-1 v1.20.11-rancher1-2 v1.20.11-rancher1-1 v1.20.10-rancher1-2 v1.20.10-rancher1-1 v1.20.9-rancher1-1 v1.20.8-rancher1-1 v1.20.6-rancher1-1 v1.20.5-rancher1-1 v1.20.4-rancher1-1 v1.19.16-rancher1-2 v1.19.16-rancher1-1 v1.19.15-rancher1-2 v1.19.15-rancher1-1 v1.19.14-rancher1-2 v1.19.14-rancher1-1 v1.19.13-rancher1-1 v1.19.12-rancher1-1 v1.19.10-rancher1-1 v1.19.9-rancher1-1 v1.19.8-rancher1-1 v1.19.7-rancher1-1 v1.19.6-rancher1-1 v1.19.4-rancher1-1 v1.19.3-rancher1-2 v1.19.3-rancher1-1 v1.19.2-rancher1-1 v1.18.20-rancher1-3 v1.18.20-rancher1-2 v1.18.20-rancher1-1 v1.18.18-rancher1-2 v1.18.18-rancher1-1 v1.18.17-rancher1-3 v1.18.17-rancher1-2 v1.18.17-rancher1-1 v1.18.16-rancher1-3 v1.18.16-rancher1-2 v1.18.16-rancher1-1 v1.18.15-rancher1-3 v1.18.15-rancher1-2 v1.18.15-rancher1-1 v1.18.14-rancher1-3 v1.18.14-rancher1-2 v1.18.14-rancher1-1 v1.18.12-rancher1-1 v1.18.10-rancher1-2 v1.18.10-rancher1-1 v1.18.9-rancher1-1 v1.18.8-rancher1-1 v1.18.6-rancher1-2 v1.18.6-rancher1-1 v1.18.3-rancher2-2 v1.18.3-rancher2-1 v1.17.17-rancher2-4 v1.17.17-rancher2-3 v1.17.17-rancher2-2 v1.17.17-rancher2-1 v1.17.17-rancher1-3 v1.17.17-rancher1-2 v1.17.17-rancher1-1 v1.17.16-rancher1-3 v1.17.16-rancher1-2 v1.17.16-rancher1-1 v1.17.14-rancher1-1 v1.17.13-rancher1-2 v1.17.13-rancher1-1 v1.17.12-rancher1-1 v1.17.11-rancher1-1 v1.17.9-rancher1-2 v1.17.9-rancher1-1 v1.17.6-rancher2-2 v1.17.6-rancher2-1 v1.17.5-rancher1-1 v1.17.4-rancher1-2 v1.17.4-rancher1-1 v1.17.3-rancher1-1 v1.17.2-rancher1-2 v1.17.2-rancher1-1 v1.17.0-rancher1-2 v1.17.0-rancher1-1 v1.16.15-rancher1-4 v1.16.15-rancher1-3 v1.16.15-rancher1-2 v1.16.15-rancher1-1 v1.16.14-rancher1-1 v1.16.13-rancher1-2 v1.16.13-rancher1-1 v1.16.10-rancher2-2 v1.16.10-rancher2-1 v1.16.9-rancher1-1 v1.16.8-rancher1-2 v1.16.8-rancher1-1 v1.16.7-rancher1-1 v1.16.6-rancher1-2 v1.16.6-rancher1-1 v1.16.4-rancher1-1 v1.16.3-rancher1-1 v1.16.2-rancher2-1 v1.16.2-rancher1-1 v1.16.1-rancher1-1 v1.15.12-rancher2-8 v1.15.12-rancher2-7 v1.15.12-rancher2-6 v1.15.12-rancher2-5 v1.15.12-rancher2-4 v1.15.12-rancher2-3 v1.15.12-rancher2-2 v1.15.12-rancher1-1 v1.15.11-rancher1-3 v1.15.11-rancher1-2 v1.15.11-rancher1-1 v1.15.11-rancher1-0 v1.15.10-rancher1-1 v1.15.9-rancher1-1 v1.15.7-rancher1-1 v1.15.6-rancher1-2 v1.15.5-rancher2-2 v1.15.5-rancher1-2 v1.15.5-rancher1-1 v1.15.4-rancher1-2 v1.15.4-rancher1-1 v1.15.3-rancher1-1 v1.15.2-rancher1-1 v1.15.0-rancher1-1 v1.14.10-rancher1-1 v1.14.10-rancher1-0 v1.14.9-rancher1-2 v1.14.9-rancher1-1 v1.14.8-rancher2-1 v1.14.8-rancher1-1 v1.14.7-rancher1-1 v1.14.6-rancher1-1 v1.14.5-rancher1-1 v1.14.3-rancher1-1 v1.14.1-rancher1-2 v1.14.1-rancher1-1 v1.13.12-rancher2-1 v1.13.12-rancher1-1 v1.13.11-rancher1-1 v1.13.10-rancher1-2 v1.13.10-rancher1-1 v1.13.9-rancher1-2 v1.13.9-rancher1-1 v1.13.7-rancher1-1 v1.13.5-rancher1-3 v1.13.5-rancher1-2 v1.13.5-rancher1-1 v1.13.4-rancher1-2 v1.13.4-rancher1-1 v1.13.1-rancher1-2 v1.13.1-rancher1-1 v1.12.10-rancher1-2 v1.12.10-rancher1-1 v1.12.9-rancher1-1 v1.12.7-rancher1-4 v1.12.7-rancher1-3 v1.12.7-rancher1-2 v1.12.7-rancher1-1 v1.12.6-rancher1-2 v1.12.6-rancher1-1 v1.12.5-rancher1-2 v1.12.5-rancher1-1 v1.12.4-rancher1-1 v1.12.3-rancher1-1 v1.12.1-rancher1-1 v1.12.0-rancher1-1 v1.11.9-rancher1-3 v1.11.9-rancher1-2 v1.11.9-rancher1-1 v1.11.8-rancher1-1 v1.11.6-rancher1-1 v1.11.5-rancher1-1 v1.11.3-rancher1-1 v1.11.2-rancher1-2 v1.11.2-rancher1-1 v1.11.1-rancher1-1 v1.10.12-rancher1-1 v1.10.11-rancher1-1 v1.10.5-rancher1-2 v1.10.5-rancher1-1 v1.10.3-rancher2-1 v1.10.1-rancher2-1 v1.10.0-rancher1-1 v1.9.7-rancher2-2 v1.9.7-rancher2-1 v1.9.5-rancher1-1 v1.8.11-rancher2-1], got v1.23.6-rancher1-1
```

**Comment by Sartigan** [comment]
When can we expect an update for this? The provider has been locked to RKE Version 1.3.3 for 7-8 months now... we are currently locked to kubernetes v1.22.4-rancher1-1 and we can not use important etcd fixes. because of the error above.

**Comment by annablender** [comment]
We are releasing terraform provider RKE v1.3.2 today which uses RKE v1.3.11 (has kubernetes default version 1.23.6 and required etcd fixes). It should be published by Hashicorp by Monday. Once that is done, this issue can be retested.

**Comment by Patricol** [comment]
I was too excited to wait after seeing the release; and used `dev_overrides` to install it. I wish I had done so much sooner, because this update fixed multiple issues that I've spent >40 hours debugging and dozens more hours working around over the last half a year.

Thank you so much for this release!

**Comment by annablender** [comment]
@Patricol wonderful! It's my pleasure to help.

**Comment by hutm** [comment]
@annablender , will this release be pushed to terraform provider registry? https://registry.terraform.io/providers/rancher/rke/latest points to 1.3.1. Thanks!

**Comment by Sartigan** [comment]
> @annablender , will this release be pushed to terraform provider registry? https://registry.terraform.io/providers/rancher/rke/latest points to 1.3.1. Thanks!

Comment says it will be pushed monday

**Comment by hutm** [comment]
@Sartigan , @annablender it seems that did not happen on Monday. Are there plans to update terraform registry release? Thanks!

**Comment by jiaqiluo** [comment]
FYI, 1.3.12 is available now. See https://registry.terraform.io/providers/rancher/rke/1.3.2

**Comment by Josh-Diamond** [comment]
Ticket #342 - Test Results [cont. ]

Failed test:

- Test provisioning with kubernetes version 1.24 

Verified with rke `v1.3.2`:
1. Provision a single-node RKE cluster w/ k8s `v1.23.6` and `enable_cri_dockerd = true`
2. Verified - Successfully provisions; healthy and ready cluster
3. Provision a single-node RKE cluster w/ k8s `v1.23.6` and `enable_cri_dockerd = false`
4. Verified- Successfully provisions; healthy and ready cluster
5. Test provisioning a three-node RKE cluster [individual roles] w/ k8s `v1.23.6`
6. Verified - Successfully provisions; healthy and ready cluster
7. Provision cluster w/ k8s 1.24
8. Veri-FAILED - `Error: expected kubernetes_version to be one of [...], got v1.24.2-rancher1-1`

Screenshots:

`Step 2 - Status`
![enabled_ready](https://user-images.githubusercontent.com/46494969/181632283-a3789e2b-01dc-4f9f-a6dc-60baa06befe4.png)

`Step 2 - Performance`
<img width="908" alt="enabled_top" src="https://user-images.githubusercontent.com/46494969/181632395-0d00ee27-e133-4919-b3b6-164c009c3d89.png">

`Step 4 - Status`
![disabled_ready](https://user-images.githubusercontent.com/46494969/181632507-fbc22aa9-361e-4736-b796-71f828a1535d.png)

`Step 4 - Performance`
<img width="913" alt="disabled_top" src="https://user-images.githubusercontent.com/46494969/181632553-7e3a99ec-1afd-40ed-bfcc-daecceefefcf.png">

`Step 6 - Status`
![3_node_ready](https://user-images.githubusercontent.com/46494969/181632791-0429ece6-f43f-4421-ac76-67147516dcb4.png)

`Step 8 - Logs`
```
│ Error: expected kubernetes_version to be one of [v1.23.6-rancher1-1 v1.23.4-rancher1-2 v1.23.4-rancher1-1 v1.22.9-rancher1-1 v1.22.7-rancher1-2 v1.22.7-rancher1-1 v1.22.6-rancher1-1 v1.22.5-rancher2-1 v1.22.5-rancher1-1 v1.22.4-rancher1-1 v1.21.12-rancher1-1 v1.21.10-rancher1-1 v1.21.9-rancher1-2 v1.21.9-rancher1-1 v1.21.8-rancher2-1 v1.21.8-rancher1-1 v1.21.7-rancher1-1 v1.21.6-rancher1-2 v1.21.6-rancher1-1 v1.21.5-rancher1-1 v1.21.4-rancher1-1 v1.20.15-rancher1-3 v1.20.15-rancher1-2 v1.20.15-rancher1-1 v1.20.14-rancher2-1 v1.20.14-rancher1-1 v1.20.13-rancher1-1 v1.20.12-rancher1-2 v1.20.12-rancher1-1 v1.20.11-rancher1-2 v1.20.11-rancher1-1 v1.20.10-rancher1-2 v1.20.10-rancher1-1 v1.20.9-rancher1-1 v1.20.8-rancher1-1 v1.20.6-rancher1-1 v1.20.5-rancher1-1 v1.20.4-rancher1-1 v1.19.16-rancher1-5 v1.19.16-rancher1-4 v1.19.16-rancher1-3 v1.19.16-rancher1-2 v1.19.16-rancher1-1 v1.19.15-rancher1-2 v1.19.15-rancher1-1 v1.19.14-rancher1-2 v1.19.14-rancher1-1 v1.19.13-rancher1-1 v1.19.12-rancher1-1 v1.19.10-rancher1-1 v1.19.9-rancher1-1 v1.19.8-rancher1-1 v1.19.7-rancher1-1 v1.19.6-rancher1-1 v1.19.4-rancher1-1 v1.19.3-rancher1-2 v1.19.3-rancher1-1 v1.19.2-rancher1-1 v1.18.20-rancher1-3 v1.18.20-rancher1-2 v1.18.20-rancher1-1 v1.18.18-rancher1-2 v1.18.18-rancher1-1 v1.18.17-rancher1-3 v1.18.17-rancher1-2 v1.18.17-rancher1-1 v1.18.16-rancher1-3 v1.18.16-rancher1-2 v1.18.16-rancher1-1 v1.18.15-rancher1-3 v1.18.15-rancher1-2 v1.18.15-rancher1-1 v1.18.14-rancher1-3 v1.18.14-rancher1-2 v1.18.14-rancher1-1 v1.18.12-rancher1-1 v1.18.10-rancher1-2 v1.18.10-rancher1-1 v1.18.9-rancher1-1 v1.18.8-rancher1-1 v1.18.6-rancher1-2 v1.18.6-rancher1-1 v1.18.3-rancher2-2 v1.18.3-rancher2-1 v1.17.17-rancher2-4 v1.17.17-rancher2-3 v1.17.17-rancher2-2 v1.17.17-rancher2-1 v1.17.17-rancher1-3 v1.17.17-rancher1-2 v1.17.17-rancher1-1 v1.17.16-rancher1-3 v1.17.16-rancher1-2 v1.17.16-rancher1-1 v1.17.14-rancher1-1 v1.17.13-rancher1-2 v1.17.13-rancher1-1 v1.17.12-rancher1-1 v1.17.11-rancher1-1 v1.17.9-rancher1-2 v1.17.9-rancher1-1 v1.17.6-rancher2-2 v1.17.6-rancher2-1 v1.17.5-rancher1-1 v1.17.4-rancher1-2 v1.17.4-rancher1-1 v1.17.3-rancher1-1 v1.17.2-rancher1-2 v1.17.2-rancher1-1 v1.17.0-rancher1-2 v1.17.0-rancher1-1 v1.16.15-rancher1-4 v1.16.15-rancher1-3 v1.16.15-rancher1-2 v1.16.15-rancher1-1 v1.16.14-rancher1-1 v1.16.13-rancher1-2 v1.16.13-rancher1-1 v1.16.10-rancher2-2 v1.16.10-rancher2-1 v1.16.9-rancher1-1 v1.16.8-rancher1-2 v1.16.8-rancher1-1 v1.16.7-rancher1-1 v1.16.6-rancher1-2 v1.16.6-rancher1-1 v1.16.4-rancher1-1 v1.16.3-rancher1-1 v1.16.2-rancher2-1 v1.16.2-rancher1-1 v1.16.1-rancher1-1 v1.15.12-rancher2-8 v1.15.12-rancher2-7 v1.15.12-rancher2-6 v1.15.12-rancher2-5 v1.15.12-rancher2-4 v1.15.12-rancher2-3 v1.15.12-rancher2-2 v1.15.12-rancher1-1 v1.15.11-rancher1-3 v1.15.11-rancher1-2 v1.15.11-rancher1-1 v1.15.11-rancher1-0 v1.15.10-rancher1-1 v1.15.9-rancher1-1 v1.15.7-rancher1-1 v1.15.6-rancher1-2 v1.15.5-rancher2-2 v1.15.5-rancher1-2 v1.15.5-rancher1-1 v1.15.4-rancher1-2 v1.15.4-rancher1-1 v1.15.3-rancher1-1 v1.15.2-rancher1-1 v1.15.0-rancher1-1 v1.14.10-rancher1-1 v1.14.10-rancher1-0 v1.14.9-rancher1-2 v1.14.9-rancher1-1 v1.14.8-rancher2-1 v1.14.8-rancher1-1 v1.14.7-rancher1-1 v1.14.6-rancher1-1 v1.14.5-rancher1-1 v1.14.3-rancher1-1 v1.14.1-rancher1-2 v1.14.1-rancher1-1 v1.13.12-rancher2-1 v1.13.12-rancher1-1 v1.13.11-rancher1-1 v1.13.10-rancher1-2 v1.13.10-rancher1-1 v1.13.9-rancher1-2 v1.13.9-rancher1-1 v1.13.7-rancher1-1 v1.13.5-rancher1-3 v1.13.5-rancher1-2 v1.13.5-rancher1-1 v1.13.4-rancher1-2 v1.13.4-rancher1-1 v1.13.1-rancher1-2 v1.13.1-rancher1-1 v1.12.10-rancher1-2 v1.12.10-rancher1-1 v1.12.9-rancher1-1 v1.12.7-rancher1-4 v1.12.7-rancher1-3 v1.12.7-rancher1-2 v1.12.7-rancher1-1 v1.12.6-rancher1-2 v1.12.6-rancher1-1 v1.12.5-rancher1-2 v1.12.5-rancher1-1 v1.12.4-rancher1-1 v1.12.3-rancher1-1 v1.12.1-rancher1-1 v1.12.0-rancher1-1 v1.11.9-rancher1-3 v1.11.9-rancher1-2 v1.11.9-rancher1-1 v1.11.8-rancher1-1 v1.11.6-rancher1-1 v1.11.5-rancher1-1 v1.11.3-rancher1-1 v1.11.2-rancher1-2 v1.11.2-rancher1-1 v1.11.1-rancher1-1 v1.10.12-rancher1-1 v1.10.11-rancher1-1 v1.10.5-rancher1-2 v1.10.5-rancher1-1 v1.10.3-rancher2-1 v1.10.1-rancher2-1 v1.10.0-rancher1-1 v1.9.7-rancher2-2 v1.9.7-rancher2-1 v1.9.5-rancher1-1 v1.8.11-rancher2-1], got v1.24.2-rancher1-1
```

**Comment by jiaqiluo** [comment]
It looks like the failure of creating 1.24 cluster is expected, because terraform-provider-rke 1.3.2 embeds rke 1.3.11 which does not support 1.24. ( upcoming rke 1.3.13 will support 1.24)

**Comment by Josh-Diamond** [comment]
Provisioning w/ k8s 1.24 will be tracked https://github.com/rancher/terraform-provider-rke/issues/354
